metrics-scraper/monitoring.go

```diff
@@ -69,16 +70,49 @@ func Start(configPath string) {
 	wg.Wait()
 }
 
+// tryConnect, try to connect to the node forever until it works
+// it'll stop when it acquire a connection or it's asked to stop
+func tryConnect(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	config NodeConfig,
+) (c *rpc.Client, err error) {
```

Comment:
Let's get rid of the named return params. They make this fn hard to read and error prone. They are also source of inconsitent behavior: if the context is done before the first iteration of the for loop, the error will be nil but after at least one iteration, it will be non-nil.

Moreover, I think we don't have to return an error from this function but rather `(*rpc.Client, bool)`. If we don't connect and the ctx is done, we return `(nil, false)`. Otherwise, `(c, true)`.

Reply:
addressed in: d1755782d3115b584b67e6fb763dc3d62386ef81